server/controllers/headers.js

```diff
@@ -0,0 +1,8 @@
+export const BggFilterPlayerCount = 'Bgg-Filter-Player-Count';
+export const BggFilterMinDuration = 'Bgg-Filter-Min-Duration';
+export const BggFilterMaxDuration = 'Bgg-Filter-Max-Duration';
+
+export function filterHeaders(requestHeaders) {
+    Object.keys(requestHeaders).forEach(key => requestHeaders[key] === undefined && delete requestHeaders[key]);

```

Comment:
I would try to avoid using the `delete` keyword where possible.  This function, currently, will mutate the object passed in, which may lead to unexpected results.

Instead I would probably recommend using something like [`Array.filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to achieve this.  That way you can filter out properties with a value of `undefined` and create a new object.

This is especially neat using [`Object.entries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries).

Something like:
```
return Object.entries(requestHeaders)
  .filter([key, val] => val !== undefined)
  .reduce((acc, [key, val]) => ({ ...acc, [key]: val }), {});
```